### PR TITLE
Sectoral constraint and detailed matrix disaggregation

### PR DESCRIPTION
**Pull request by isaac-tfn**
Changes
---------
A new furnessing method which allows constraint to a more aggregate matrix. This is envisaged as being used for disaggregating matrices, where the original trip ends are disagreggated using caf.space/caf.toolkit, seed values generated using original cost function params, and the matrix being controlled back to the original, more aggregate matrix during furnessing.

It is designed to work where the new, disaggregate, zone system nests perfectly within the original zone system; i.e. translation factors from disaggregate to aggregate are all 1.

The method for disaggregating matrices is also included in this PR.

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/distribute/matrix_translation.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import pandas as pd
```
Comment (MattBuckley-TfN) [inline src/caf/distribute/matrix_translation.py:1]: I think this would be useful moved to an examples folder in the repo root. Also, could you include some docstrings and type annotations

**Review by MattBuckley-TfN — CHANGES_REQUESTED** [review]
Overall I think the code looks like it works just some comments around tidying up and documentation.

- Merge main into this branch to resolve conflicts.
- Isort needs running
- Tests failing due to type annotations with '|', should be fixed by adding `from __future__ import annotations` to `test_multi_area.py` and `test_single_area.py`.
- Quite a few pylint issues to deal with or add ignore comments
